tap_hubspot_beta/tap.py

```diff
@@ -187,6 +197,53 @@ def catalog_dict(self) -> dict:
                         stream["schema"]["properties"][field]["field_meta"] = stream_class.fields_metadata.get(field, {})
         return catalog
 
+    def generate_stream_class(self, custom_object: Dict[str, Any]) -> hubspotV3Stream:
+        # check for required fields to construct the custom objects class
+        _id = custom_object.get("id")
+        if not _id:
+            raise ValueError(f"Missing id in custom object. object={custom_object}.")
+        name = custom_object.get("name")
+        if not name:
+            raise ValueError(f"Missing name in custom object. id={_id}.")
+        object_type_id = custom_object.get("objectTypeId")
+        if not object_type_id:
+            raise ValueError(f"Missing objectTypeId in custom object. id={_id}. name={name}")
+        properties = custom_object.get("properties")
+        if not properties:
+            raise ValueError(f"Missing properties in custom object. id={_id}. name={name}. object_type_id={object_type_id}")
+        
+        if custom_object.get("archived", False):
+            name = "archived_" + name
+        class_name = name + "_Stream"
+        class_name = "".join(word.capitalize() for word in class_name.split("_"))
+
+        self.logger.info(f"Creating class {class_name}")
+
+        return type(
+            class_name,
+            (hubspotV3Stream,),
+            {
+                "name": name,
+                "path": f"crm/v3/objects/{object_type_id}/",
+                "records_jsonpath": "$.results[*]",
+                "primary_keys": ["id"],
+                "replication_key": "updatedAt",
+                "page_size": 100,
+                "schema": self.generate_schema(properties),
+            },
+        )
+
+    def generate_schema(self, properties: List[Dict[str, Any]]) -> dict:
+        properties_list = [th.Property("id", th.StringType), th.Property("updatedAt", th.DateTimeType)]
+        for property in properties:
+            field_name = property.get("name")
+            if not field_name:
+                self.logger.info(f"Skipping field without name.")
```

Comment:
This is odd... Why would we ignore properties that don't have the `name` field?

Reply:
because withou a name, we cannot build the property in PropertiesList